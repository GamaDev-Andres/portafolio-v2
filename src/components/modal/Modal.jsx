import ReactDOM from 'react-dom'

export const ModalPortal = ({ children, closeModal }) => {
  return ReactDOM.createPortal(
    <div
      className="flex items-center fixed justify-center z-50 w-screen h-screen bg-fondoModal"
      onClick={() => closeModal()}
    >
      <div
        className="relative flex items-center justify-center mx-auto"
        onClick={(e) => e.stopPropagation()}
      >
        <span
          className="absolute top-2 right-2 cursor-pointer mx-auto p-1 text-red-600"
          title="cerrar"
          onClick={closeModal}
        >
          <i className="fas fa-times-circle"></i>
        </span>
        {children}
      </div>
    </div>,
    document.getElementById('root-modal')
  )
}
