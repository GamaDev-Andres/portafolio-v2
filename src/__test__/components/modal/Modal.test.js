/**
 * @jest-environment jsdom
 */
import Enzyme from 'enzyme'
import { ModalPortal } from '../../../components/modal/Modal'

describe('probando <ModalPortal/>', () => {
  const modalRoot = global.document.createElement('div')
  modalRoot.setAttribute('id', 'root-modal')
  const body = global.document.querySelector('body')
  body.appendChild(modalRoot)
  const jestFn = jest.fn()
  const wrapper = Enzyme.mount(
    <ModalPortal closeModal={jestFn} >
      <div id="hijo">soy un hijo</div>
    </ModalPortal>
  )

  const rootModal = document.querySelector('#root-modal')
  test('debe montarse correctamente en "#root-modal"', () => {
    expect(wrapper).toMatchSnapshot()
    expect(rootModal.hasChildNodes()).toBe(true)
  })
  test('debe mostrar el hijo enviado correctamente en "#root-modal"', () => {
    expect(wrapper.find('div#hijo').text().trim()).toBe('soy un hijo')
  })
  test('debe llamar a la funcion "close" correctamente al dark click en el wrapper', () => {
    wrapper.find('div').at(0).simulate('click')
    expect(jestFn).toHaveBeenCalled()
  })
  test('debe llamar a la funcion "close" correctamente al dark click en el span', () => {
    wrapper.find('span').at(0).simulate('click')
    expect(jestFn).toHaveBeenCalled()
  })
})
