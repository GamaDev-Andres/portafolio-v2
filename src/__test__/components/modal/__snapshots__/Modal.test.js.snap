// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`probando <ModalPortal/> debe montarse correctamente en "#root-modal" 1`] = `
<Portal
  containerInfo={
    <div
      id="root-modal"
    >
      <div
        class="flex items-center fixed justify-center z-50 w-screen h-screen bg-fondoModal"
      >
        <div
          class="relative flex items-center justify-center mx-auto"
        >
          <span
            class="absolute top-2 right-2 cursor-pointer mx-auto p-1 text-red-600"
            title="cerrar"
          >
            <i
              class="fas fa-times-circle"
            />
          </span>
          <div
            id="hijo"
          >
            soy un hijo
          </div>
        </div>
      </div>
    </div>
  }
>
  <div
    className="flex items-center fixed justify-center z-50 w-screen h-screen bg-fondoModal"
    onClick={[Function]}
  >
    <div
      className="relative flex items-center justify-center mx-auto"
      onClick={[Function]}
    >
      <span
        className="absolute top-2 right-2 cursor-pointer mx-auto p-1 text-red-600"
        onClick={[MockFunction]}
        title="cerrar"
      >
        <i
          className="fas fa-times-circle"
        />
      </span>
      <div
        id="hijo"
      >
        soy un hijo
      </div>
    </div>
  </div>
</Portal>
`;
